core/unix/signal.c

```diff
@@ -4807,6 +4807,28 @@ master_signal_handler_C(byte *xsp)
         return;
     }
 #endif
+
+    /* We are dropping asynchronous signals during detach. The thread may already have
+     * lost its TLS (xref i#3535). A safe read may result into a race affecting
+     * asynchronous non-alarm signals between delivering the SIGSEGV and restoring the
+     * app's signal handlers.
+     *
+     * Note that besides dropping potentially important signals, there is still a small
+     * race window if the signal gets delivered after the detach dhas finished, i.e.
+     * doing detach is false. This is an issue in particular if the app has started
+     * re-attaching.
+     *
+     * Signals that are no clearly asynchronous may hit corner case(s) of i#3535.
+     * (xref i#26).
+     */
+    if (doing_detach && can_always_delay[sig]) {
+        DOLOG(1, LOG_ASYNCH, { dump_sigcontext(GLOBAL_DCONTEXT, sc); });
+        SYSLOG_INTERNAL_ERROR("ERROR: master_signal_handler with unreliable dcontext "
```

Comment:
Please mention detach in the message.
Also mention that we're dropping this signal (and continuing).

Reply:
I thought SYSLOG_INTERNAL_ERROR causes an immediate fail and therefore we're not continuing, but I guess this goes to the log and we do continue.